pkg/controller/installation/validation.go

```diff
@@ -344,6 +344,10 @@ func validateCustomResource(instance *operatorv1.Installation) error {
 		if _, ok := validComponentNames[resource.ComponentName]; !ok {
 			return fmt.Errorf("Installation spec.ComponentResources.ComponentName %s is not supported", resource.ComponentName)
 		}
+
+		if _, found := resource.Labels["k8s-app"]; found {
```

Comment:
I think we'll also need to validate that the given label keys and values are valid Kubernetes label syntax.

Reply:
Might be doable via kubebuilder? https://book.kubebuilder.io/reference/markers/crd-validation.html

If not, then we should add the check here.

Reply:
1) kubebuilder does not seem to have a way to validate the map[string]string.

2) If I try to create a non conform label, it is already logged:
`{"level":"error","ts":1647979877.3085115,"logger":"controller_installation","msg":"Error creating / updating resource","Request.Namespace":"tigera-operator","Request.Name":"node-certs","error":"Deployment.apps \"calico-typha\" is invalid: [spec.template.labels: Invalid value: \"_batch.kubernetes.io/job-completion-index\": prefix part a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*'), spec.template.labels: Invalid value: \"3/\": a valid label must be an empty string or consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyValue',  or 'my_value',  or '12345', regex used for validation is '(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])?')]","stacktrace":"sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/home/ioannis/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.7/pkg/internal/controller/controller.go:253\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\t/home/ioannis/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.7/pkg/internal/controller/controller.go:214"}
`

Writing an extra check will not help, in order to throw an error on `kubectl apply -f`

Reply:
Yeah, the benefit of doing it here would be to prevent us attempting to write invalid resource to the API server and giving us an opportunity to provide a more targeted error message to the user via `TigeraStatus` - both relatively minor things but probably worth doing. 

We try to avoid sending requests to the apiserver that we know are going to fail

Reply:
It sounds reasonable. I pushed the validation on the branch